inlang/packages/paraglide/paraglide-js/src/compiler/runtime/localize-url.js

```diff
@@ -212,6 +211,39 @@ function deLocalizeUrlDefaultPattern(url) {
 	return urlObj;
 }
 
+/**
+ * Takes matches of implicit wildcards in the UrlPattern (when a part is missing
+ * it is equal to '*') and adds them back to the result of fillPattern.
+ *
+ * At least protocol and hostname are required to create a valid URL inside fillPattern.
+ *
+ * @param {URL} url
+ * @param {any} match
+ * @returns {URL}
+ */
+function fillMissingUrlParts(url, match) {
+	if (match.username.groups["0"]) {
+		url.username = match.username.groups["0"] ?? "";
+	}
+	if (match.password.groups["0"]) {
+		url.password = match.password.groups["0"] ?? "";
+	}
+	if (match.port.groups["0"]) {
+		url.port = match.port.groups["0"] ?? "";
+	}
+	if (match.pathname.groups["0"]) {
+		url.pathname = match.pathname.groups["0"] ?? "";
+	}
+	if (match.search.groups["0"]) {
+		url.search = match.search.groups["0"] ?? "";
+	}
+	if (match.hash.groups["0"]) {
+		url.hash = match.hash.groups["0"] ?? "";
+	}
+
+	return url;
```

Comment:
This will probably lead to a follow up issue/PR where people are asking for all wildcards to be filled in but it's good enough for now.